nomad/node_endpoint_test.go

```diff
@@ -901,6 +901,97 @@ func TestClientEndpoint_Drain_Down(t *testing.T) {
 	})
 }
 
+func TestClientEndpoint_UpdateEligibility(t *testing.T) {
```

Comment:
Can we add a test for the guard against making a draining node eligible for scheduling? That's a pretty important case to test as if we accidently break it, drains are very very broken.

Reply:
Never mind! I see the test below. Love the indepth validation around this critical invariant.